llvm/lib/Target/AMDGPU/SIISelLowering.cpp

```diff
@@ -16142,11 +16142,8 @@ static bool atomicIgnoresDenormalModeOrFPModeIsFTZ(const AtomicRMWInst *RMW) {
 
 static OptimizationRemark emitAtomicRMWLegalRemark(const AtomicRMWInst *RMW) {
   LLVMContext &Ctx = RMW->getContext();
-  SmallVector<StringRef> SSNs;
-  Ctx.getSyncScopeNames(SSNs);
-  StringRef MemScope = SSNs[RMW->getSyncScopeID()].empty()
-                           ? "system"
-                           : SSNs[RMW->getSyncScopeID()];
+  StringRef SS = Ctx.getSyncScopeName(RMW->getSyncScopeID());
```

Comment:
This returns optional, but this is directly using it as StringRef?

FWIW I think having this return an empty StringRef in the unknown cases is a reasonable behavior vs. optional. I would expect unrecognized names to be treated as system

Reply:
Have updated AMDGPU to map an out of bound `SyncScope::ID` to `""` (system) with `.value_or("")`, but NVPTX maps out of bounds `SyncScope::ID` to errors, so just returning an empty string doesn't work there.